tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -17,18 +17,33 @@ import com.microsoft.z3.enumerations.Z3_lbool
 import scala.collection.mutable
 
 /**
-  * An implementation of a SolverContext using Z3.
+  * <p>
+  * An implementation of a SolverContext using Z3. Note that this class overrides the global z3 settings
+  * `sat.random_seed`, `smt.random_seed`, `fp.spacer.random_seed`, and `sls.random_seed` with `config.randomSeed`.
+  * Although it is usually not a good idea to override the global settings, we do it to isolate the code
+  * specific to z3 in this class.
+  * </p>
   *
   * @author Igor Konnov
   */
-class Z3SolverContext(debug: Boolean = false, profile: Boolean = false) extends SolverContext {
+class Z3SolverContext(config: SolverConfig) extends SolverContext {
   private val id: Long = Z3SolverContext.createId()
 
   /**
     * A log writer, for debugging purposes.
     */
   private val logWriter: PrintWriter = initLog()
 
+  // dump the configuration parameters in the log
+  // set the global configuration parameters for z3 modules
+  Z3SolverContext.RANDOM_SEED_PARAMS.foreach {
+    p =>
+      Global.setParameter(p, config.randomSeed.toString)
+      logWriter.println(";; %s = %s".format(p, config.randomSeed))
+//    the following fails with an exception: java.lang.NoSuchFieldError: value
+//      logWriter.println(";; %s = %s".format(p, Global.getParameter(p)))
```

Comment:
Is there value to this comment? Looks me like it might just be a left over...

Reply:
Right. I wanted to print out the value as reported by z3 itself. But the current version produces an exception. So I left it for the future, or just as documentation for the future if someone (like me) wants to write the same code again :-)